index.js

```diff
@@ -60,6 +69,19 @@ module.exports = function (options, logger) {
       obj.headers = headers
     }
 
+    if (excludeHeaders && obj.headers) {
+      var headers = {}
+      Object.keys(obj.headers).forEach(function(name) {
+        headers[name] = obj.headers[name]
+      })
```

Comment:
Good shout. I've made a further change to do all the copying/deleting in a single `reduce` loop. What do you think?